SVT: Liberty Tools start failure observed using new liberty-gradle-plugin with newly supported Gradle 9 due to -b command line option

SVT: Liberty Tools start failure observed using new liberty-gradle-plugin with newly supported Gradle 9 due to -b command line option

I've downloaded the latest VSCode on my Windows 11 laptop.
```
Version: 1.113.0
Commit: cfbea10c5ffb233ea9177d34726e6056e89913dc
Date: 2026-03-24T15:07:18+01:00
Electron: 39.8.3
ElectronBuildId: 13620978
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Windows_NT x64 10.0.26100
```

I have a simple wcaGradleApp with the following build.gradle contents that have now been updated to use the new 4.0.0 version (released early March) for liberty-gradle-plugin (https://central.sonatype.com/artifact/io.openliberty.tools/liberty-gradle-plugin) with Gradle 9.3.1.

```
buildscript {
    repositories {
        mavenCentral()
        maven {
            url = 'https://central.sonatype.com/repository/maven-snapshots/'
        }
    }
    dependencies {
        classpath 'io.openliberty.tools:liberty-gradle-plugin:4.0.0'
    }
}

plugins {
    id 'war'
}
apply plugin:'liberty'

repositories {
    mavenCentral()
}

dependencies {
    providedCompile 'jakarta.servlet:jakarta.servlet-api:5.0.0' 
}

war {
  archiveFileName = 'wcaGradleApp.war'
}

clean.dependsOn 'libertyStop'
```

I've installed the latest Liberty Tools extension from Marketplace which shows version 25.0.11.

Using VSCode, when I select 'Liberty Dashboard' -> app -> Start, I observe the following failure in the Terminal.
```
PS C:\Users\XXX\git\wcaGradleApp\app> & "c:\Users\XXX\git\wcaGradleApp\gradlew.bat" libertyDev -b="c:\Users\XXX\git\wcaGradleApp\app\build.gradle"
FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :app:libertyDev from command line.
> Unknown command-line option '-b'.

* Try:
> Run gradlew help --task :app:libertyDev to get task usage details.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 35s
```

Previous version of this app using liberty-gradle-plugin version 3.9.5 with Gradle 8.4 worked successfully as the -b command line option was known.

--- a/src/util/commandUtils.ts
+++ b/src/util/commandUtils.ts
@@ -1,6 +1,6 @@
 /*
  * IBM Confidential
- * Copyright IBM Corp. 2024
+ * Copyright IBM Corp. 2024, 2026
  */
 
 import * as Path from "path";
@@ -52,16 +52,16 @@ export async function getCommandForMaven(pomPath: string, command: string, termi
  */
 export async function getCommandForGradle(buildGradlePath: string, command: string, terminalType?: String, customCommand?: string): Promise<string> {
     let gradleCmdStart = await gradleCmd(buildGradlePath);
+    const projectDir = Path.dirname(buildGradlePath);
 
     if (gradleCmdStart === "gradle") {
-        return formDefaultCommand(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+        return formGradleCommand(projectDir, "gradle", command, customCommand);
     }
     //checking the OS type for command customization
     if (isWin()) {
-        return getGradleCommandForWin(gradleCmdStart, buildGradlePath, command, terminalType, customCommand);
+        return getGradleCommandForWin(gradleCmdStart, projectDir, command, terminalType, customCommand);
     } else {
-        gradleCmdStart = Path.join(gradleCmdStart, "gradlew");
-        return formDefaultCommandWithPath(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+        return formGradleCommand(projectDir, "./gradlew", command, customCommand);
     }
 }
 
@@ -158,20 +158,17 @@ async function getLocalMavenWrapperDir(projectFolder: string): Promise<string | 
 /**
  * Returns the gradle command for windows OS based on the terminal configured
  */
-function getGradleCommandForWin(gradleCmdStart: string, buildGradlePath: string, command: string, terminalType?: String, customCommand?: string): string {
+function getGradleCommandForWin(gradleCmdStart: string, projectDir: string, command: string, terminalType?: String, customCommand?: string): string {
     switch (terminalType) {
         case ShellType.GIT_BASH:
-            gradleCmdStart = Path.join(gradleCmdStart, "gradlew");
-            return formDefaultCommandWithPath(gradleCmdStart, buildGradlePath, command, "-b=", customCommand); //Bash
+            return formGradleCommand(projectDir, "./gradlew", command, customCommand); //Bash
         case ShellType.POWERSHELL:
-            gradleCmdStart = Path.join(gradleCmdStart, "gradlew.bat");
-            return formPowershellCommand(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+            return formGradlePowershellCommand(projectDir, "gradlew.bat", command, customCommand);
         case ShellType.WSL:
-            return formLinuxBasedCommand(toDefaultWslPath(gradleCmdStart), command, "./gradlew ", customCommand); //Wsl
+            return formGradleCommand(toDefaultWslPath(projectDir), "./gradlew", command, customCommand); //Wsl
         default:
             // The default case is ShellType CMD or OTHERS
-            gradleCmdStart = Path.join(gradleCmdStart, "gradlew.bat");
-            return formDefaultCommandWithPath(gradleCmdStart, buildGradlePath, command, "-b=", customCommand);
+            return formGradleWindowsCommand(projectDir, "gradlew.bat", command, customCommand);
     }
 }
 
@@ -233,6 +230,38 @@ function formDefaultCommandWithPath(projectPath: string, buildFilePath: String, 
         return "\"" + projectPath + "\" " + `${command}` + ` ${customCommand}` + ` ${cmdOption}"${buildFilePath}"`;
     }
     return "\"" + projectPath + "\" " + `${command}` + ` ${cmdOption}"${buildFilePath}"`;
+}
+
+/**
+ * Returns the Gradle command by cd'ing into the project directory
+ * This approach is compatible with all Gradle versions including Gradle 9+
+ * where the -b option was removed
+ */
+function formGradleCommand(projectDir: string, gradleCmd: string, command: string, customCommand?: string): string {
+    if (customCommand) {
+        return `cd "${projectDir}" && ${gradleCmd} ${command} ${customCommand}`;
+    }
+    return `cd "${projectDir}" && ${gradleCmd} ${command}`;
+}
+
+/**
+ * Returns the Gradle command for PowerShell by cd'ing into the project directory
+ */
+function formGradlePowershellCommand(projectDir: string, gradleCmd: string, command: string, customCommand?: string): string {
+    if (customCommand) {
+        return `cd "${projectDir}"; .\\${gradleCmd} ${command} ${customCommand}`;
+    }
+    return `cd "${projectDir}"; .\\${gradleCmd} ${command}`;
+}
+
+/**
+ * Returns the Gradle command for Windows CMD by cd'ing into the project directory
+ */
+function formGradleWindowsCommand(projectDir: string, gradleCmd: string, command: string, customCommand?: string): string {
+    if (customCommand) {
+        return `cd "${projectDir}" && ${gradleCmd} ${command} ${customCommand}`;
+    }
+    return `cd "${projectDir}" && ${gradleCmd} ${command}`;
 }
 
 /**

